tests/views_tests.py

```diff
@@ -21,3 +22,26 @@ def test_user_update(client, user):
     assert response.status_code == 200
     assert deserialize(response)["username"] == "testtest"
     assert deserialize(response)["email"] == "something@example.com"
+
+
+def test_user_search(client, django_user_model):
+    django_user_model.objects.create(username="test", email="test@example.com", date_joined=datetime.fromisoformat("2021-01-01T00:00:00+00:00"))
+    django_user_model.objects.create(username="test2", email="test2@example.com", date_joined=datetime.fromisoformat("2021-02-01T00:00:00+00:00"))
+    django_user_model.objects.create(username="test3", email="test3@test.com", date_joined=datetime.fromisoformat("2021-03-01T00:00:00+00:00"))
+    
+    # search by username -- filter ignored
+    response = client.get(f"/users/?username=test2")
+    assert len(deserialize(response)["users"]) == 3
+
+    # search by email
+    response = client.get(f"/users/?email=test@example.com")
+    assert len(deserialize(response)["users"]) == 1
+
+    # search string
+    response = client.get(f"/users/?q=example.com")
+    assert len(deserialize(response)["users"]) == 2
+    
+    # search by date joined
+    response = client.get(f"/users/?date_joined__gte=2021-02-01T00:00:00Z&date_joined__lte=2021-02-15T00:00:00Z")
```

Comment:
@regiscamimura don't these have to be camel case? `dateJoined__gte`?

Reply:
good catch, will convert those to camel case so we know it works when passing camel cased params, but it will work with snake case because the first thing it does is to convert from camel to snake

Reply:
Seems like we support both camel and snake already on fields, so I guess we should test both, or at least camel, given that's the variant we use.